src/chainlit/lc/callbacks.py

```diff
@@ -121,9 +172,11 @@ def start_stream(self):
         )
         self.stream = streamed_message
 
-    def send_token(self, token: str):
-        if self.stream:
-            run_sync(self.stream.stream_token(token))
+    def send_token(self, token: str, final: bool = False):
+        stream = self.final_stream if final else self.stream
+        if stream:
+            run_sync(stream.stream_token(token))
```

Comment:
if final is true then we want to also call .send() on the message, since we want to persist it. Persistence is not called on each streamed token (it would flood the db)

Reply:
something like this? 11b58868b38b38157ab82f842fa1a4106cc4bd94

Reply:
yes